Library/Homebrew/cmd/outdated.rb

```diff
@@ -22,43 +24,62 @@ def outdated_args
       flag   "--json",
              description: "Print output in JSON format. Currently the default and only accepted "\
                           "value for <version> is `v1`. See the docs for examples of using the JSON "\
-                          "output: <https://docs.brew.sh/Querying-Brew>"
+                          "output: <https://docs.brew.sh/Querying-Brew>. "\
+                          "By default, this option treats all arguments as formulae. "\
+                          "To treat arguments as casks, use the --cask-only option."
       switch "--fetch-HEAD",
              description: "Fetch the upstream repository to detect if the HEAD installation of the "\
                           "formula is outdated. Otherwise, the repository's HEAD will only be checked for "\
                           "updates when a new stable or development version has been released."
+      switch "--greedy",
+             description: "Print outdated casks with `auto_updates` or `version :latest`"
+      switch "--formula-only",
+             description: "Treat all arguments as formulae"
+      switch "--cask-only",
+             description: "Treat all arguments as casks"
       switch :debug
       conflicts "--quiet", "--verbose", "--json"
+      conflicts "--formula-only", "--cask-only"
     end
   end
 
   def outdated
     outdated_args.parse
 
-    formulae = if args.resolved_formulae.blank?
-      Formula.installed
-    else
-      args.resolved_formulae
-    end
     if args.json
       raise UsageError, "invalid JSON version: #{args.json}" unless ["v1", true].include? args.json
+    end
 
-      outdated = print_outdated_json(formulae)
+    if args.formula_only? || !args.cask_only? && args.json
```

Comment:
```suggestion
    if (args.formula_only? || !args.cask_only?) && args.json
```

(assuming that's correct)

Reply:
I think without parentheses is correct because, with parentheses, `brew outdated --formula-only` would cause the conditional to be false and end up printing formulae and casks

Reply:
Sorry, I mean: can you add parentheses somewhere here 😁

Reply:
Oh, woops. Anyways, I ended up redoing the logic to be more clear because I found it to be a bit hard to understand myself